components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -40,12 +40,13 @@ import mozilla.components.service.glean.utils.parseISOTimeString
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.ktx.android.content.isMainProcess
 import mozilla.components.support.utils.ThreadUtils
+import kotlin.properties.Delegates
 
 @Suppress("TooManyFunctions", "LargeClass")
 open class GleanInternalAPI internal constructor () {
     private val logger = Logger("glean/Glean")
 
-    private var applicationContext: Context? = null
+    private var applicationContext: Context by Delegates.notNull()
```

Comment:
`lateinit var` instead?

Reply:
Just to make sure I understand: why is a change required on this line in the first place?

Reply:
We needed to cache the context to be able to use it in `cancelPingWorkers()`

Reply:
I could have used an `applicationContext?.let` but I thought if someone has forgotten to call `initialize` and then `cancelPingWorkers ` was called it would've skipped doing the actual work silently wasn't flagging an exception that applicationContext was not initialized.